File: app/src/main/java/com/kickstarter/libs/preferences/IntPreferenceType.java
package com.kickstarter.libs.preferences;

public interface IntPreferenceType {
  /**
   * Get the current value of the preference.
   */
  int get();

  /**
   * Returns whether a value has been explicitly set for the preference.
   */
  boolean isSet();

  /**
   * Set the preference to a value.
   */
  void set(int value);

  /**
   * Delete the currently stored preference.
   */
  void delete();
}

File: app/src/main/java/com/kickstarter/libs/preferences/IntPreferenceType.kt
package com.kickstarter.libs.preferences

interface IntPreferenceType {
    /**
     * Get the current value of the preference.
     */
    fun get(): Int

    /**
     * Returns whether a value has been explicitly set for the preference.
     */
    val isSet: Boolean

    /**
     * Set the preference to a value.
     */
    fun set(value: Int)

    /**
     * Delete the currently stored preference.
     */
    fun delete()
}
